components/application-operator/pkg/application-controller/reconciler_test.go

```diff
@@ -444,6 +449,43 @@ func TestApplicationReconciler_Reconcile(t *testing.T) {
 		managerClient.AssertExpectations(t)
 		ApplicationReleaseManager.AssertExpectations(t)
 	})
+
+	t.Run("should invoke release manager upgrade if application override labels has changed", func(t *testing.T) {
+		// given
+		namespacedName := types.NamespacedName{
+			Name: applicationName,
+		}
+
+		managerClient := &mocks.ApplicationManagerClient{}
+		managerClient.On(
+			"Get", context.Background(), namespacedName, mock.AnythingOfType("*v1alpha1.Application")).
+			Run(setupAppInstance).Return(nil)
+		managerClient.On("Update", context.Background(), mock.AnythingOfType("*v1alpha1.Application")).
+			Run(statusChecker.checkStatus).Return(nil)
+
+		ApplicationReleaseManager := &helmmocks.ApplicationReleaseManager{}
+		ApplicationReleaseManager.On("CheckReleaseExistence", applicationName).Return(true, nil)
+		ApplicationReleaseManager.On("CheckReleaseStatus", applicationName).Return(releaseStatus, statusDescription, nil)
+
+		ofType := mock.AnythingOfType("*v1alpha1.Application")
+		ApplicationReleaseManager.On("UpgradeApplicationRelease", ofType)
+
+		reReconciler := NewReconciler(managerClient, ApplicationReleaseManager, logger)
+
+		request := reconcile.Request{
+			NamespacedName: namespacedName,
+		}
+
+		// when
+		result, err := reReconciler.Reconcile(request)
+
+		// then
+		assert.NoError(t, err)
+		assert.NotNil(t, result)
+		managerClient.AssertExpectations(t)
+		ApplicationReleaseManager.AssertExpectations(t)
+		ApplicationReleaseManager.AssertCalled(t, "UpgradeApplicationRelease", ofType)
```

Comment:
This is not needed AssertExpectations already asserts that everything specified with On and Return was
in fact called as expected.